internal/application/application.go

```diff
@@ -377,3 +318,50 @@ func Logs(ctx context.Context, logChan chan tailer.ContainerLogLine, wg *sync.Wa
 	logger.Info("fetch")
 	return tailer.FetchLogs(ctx, logChan, wg, config, cluster)
 }
+
+// fetch is a common helper for Lookup and List. It fetches all
+// information about an application from the cluster.
+func fetch(ctx context.Context, cluster *kubernetes.Cluster, app *models.App) error {
```

Comment:
I feel this should have the `models.App` as a receiver (iow, be a method of), after all, all it does it mutating it.

Reply:
I will look into that. About to push the rebase.

Reply:
Ah. `App` is a non-local type. Making the change implies moving `fetch` into the `models` package.

Reply:
And if we do that we will get a circular dependency because the function uses methods from `application`, and is used in `application`. Verdict: Declined.